### PR TITLE
Add option to also publish Protobuf version of results from backend

### PR DESCRIPTION
**Pull request by srimanachanta**
Allows logging software and live data view to see results. Also removes the requirement for AScope to keep up with the packet serde schema and instead just use the Protobuf descriptor.

### CODE REVIEW

File: photon-client/src/components/settings/NetworkingCard.vue [hunk 1]
```diff
@@ -38,9 +40,31 @@ const isValidHostname = (v: string | undefined) => {
   return hostnameRegex.test(v);
 };
 
+const settingsHaveChanged = (): boolean => {
+  const a = useSettingsStore().network;
+  const b = tempSettingsStruct.value;
+
+  return (
```
Comment (mcm001) [inline photon-client/src/components/settings/NetworkingCard.vue:47]: Why can't we use === here?

Comment (srimanachanta) [reply photon-client/src/components/settings/NetworkingCard.vue:47]: Object equality checks are very bad in js and is a likely risk, better to just do this. Its the same way we compare resolutions manually in the frontend

File: photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java [hunk 2]
```diff
@@ -164,6 +165,9 @@ public void setConfig(NetworkConfig config) {
         } else {
             setClientMode(config.ntServerAddress);
         }
+
+        VisionModuleManager.getInstance().setShouldPublishProto(config.shouldPublishProto);
```
Comment (mcm001) [inline photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java:169]: Why not have NtDataPublisher pull directly from ConfigManager.getInstance().getConfig().getNetworkConfig().shouldPublishProto? Why this misdirection?

File: photon-targeting/src/main/java/org/photonvision/targeting/proto/MultiTargetPNPResultProto.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    @Override
+    public ProtobufMultiTargetPNPResult createMessage() {
+        return ProtobufMultiTargetPNPResult.newInstance();
+    }
+
+    @Override
+    public MultiTargetPNPResult unpack(ProtobufMultiTargetPNPResult msg) {
+        return new MultiTargetPNPResult(
+                PNPResult.proto.unpack(msg.getEstimatedPose()),
+                // TODO better way of doing this
```
Comment (mcm001) [inline photon-targeting/src/main/java/org/photonvision/targeting/proto/MultiTargetPNPResultProto.java:55]: What kind of better way do you want?

**Review by mcm001 — APPROVED** [review]

File: photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java (hunk 2 above)
Comment (srimanachanta) [reply photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java:169]: Uhhhh, me no do photon-core. I didn't know that was a thing.

File: photon-client/src/components/settings/NetworkingCard.vue (hunk 1 above)
Comment (mcm001) [reply photon-client/src/components/settings/NetworkingCard.vue:47]: 💀

File: photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java (hunk 2 above)
Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/common/dataflow/networktables/NetworkTablesManager.java:169]: In general, pipelines pulling settings from global networking settings isn't really a thing we commonly do. Usually that's all held per-pipeline. But making this a per-pipeline setting seems like also a bad idea. So I think my proposal makes more sense

**Comment by mcm001** [comment]
Tested with beta 4 glass, works for me!

**Pull request merged**